signer/signer.go

```diff
@@ -32,15 +32,25 @@ type Subject struct {
 	Names []csr.Name `json:"names"`
 }
 
+// Extension represents a raw extension to be included in the certificate.  The
+// "value" field must be Base64 encoded, using the URLEncoding variant (i.e.,
```

Comment:
Re: Encoding, I think hex would actually be nice.  For those schooled in the DER arts, it can actually be intelligible.

Re: CSR vs. SignRequest: The upshot of this change is that extensions in the CSR are ignored except for the ones processed by `ParseCertificateRequest` (primarily subjectAltName).  So the calling code is responsible for moving any extensions it wants added into the SignRequest.  I think that's correct because it means that the extension data that goes into the cert is provided by the calling code, not the entity submitting the CSR.  I'll find a place to add a comment.